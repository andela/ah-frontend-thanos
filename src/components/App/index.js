import React from 'react';
import Notification from 'react-notify-toast';
import { Router, Route, Switch } from 'react-router-dom';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faSearch } from '@fortawesome/free-solid-svg-icons';
import history from '../../commons/history';
import Home from '../Home';
import LoginPage from '../../containers/LoginPage';
import FooterConnected from '../../containers/Footer';
import Header from '../../containers/Header';
import SignUpPageConnected from '../../containers/SignUpPage';
import Articles from '../../containers/Articles';
import ArticlePageConnected from '../../containers/ArticlePage';
import ProfileConnected from '../../containers/profiles/profiles';
import EditProfilePageConnected from '../../containers/profiles/editProfile';
import PasswordResetPage from '../../containers/PasswordResetPage';
import NewPasswordPage from '../../containers/PasswordResetPage/newpasswordPage';
import CreateArticlePage from '../../containers/CreateArticlePage';
import Tags from '../TagList/viewTags';
import FollowUnfollow from '../../containers/FollowUnfollow';
import SearchPage from '../../containers/SearchPage';

library.add(faSearch);
const App = () => (
  <Router history={history}>
    <React.Fragment>
      <Header history={history} />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route path="/login" component={LoginPage} />
        <Route path="/signup" component={SignUpPageConnected} />
        <Route path="/articles" component={Articles} />
        <Route path="/article/:articleId" component={ArticlePageConnected} />
        <Route path="/profile" component={ProfileConnected} />
        <Route path="/profiles/edit" component={EditProfilePageConnected} />
        <Route path="/passwordreset" component={PasswordResetPage} />
        <Route path="/newpassword" component={NewPasswordPage} exact />
        <Route path="/createArticle" component={CreateArticlePage} />
        <Route path="/tags" component={Tags} />
        <Route exact path="/profiles/:username" component={FollowUnfollow} />
        <Route path="/search" component={SearchPage} />
      </Switch>
      <Notification />
      <FooterConnected />

    </React.Fragment>
  </Router>
);

export default App;
