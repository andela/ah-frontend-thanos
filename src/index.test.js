import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import App from './components/App';

describe('Provider and App', () => {
  let app;
  let provider;
  let store;

  beforeEach(() => {
    const middlewares = [thunk];
    const mockStore = configureStore(middlewares);
    const initialState = { article: {}, user: {} };
    store = mockStore(initialState);
    provider = shallow(<Provider store={store}><App /></Provider>);
    app = shallow(<App />);
  });

  it('renders <Provider/> correctly', () => {
    expect(provider).toMatchSnapshot();
  });

  it('renders <App/> correctly', () => {
    expect(app).toMatchSnapshot();
  });
});
