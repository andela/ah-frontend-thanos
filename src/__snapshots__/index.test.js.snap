// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provider and App renders <Provider/> correctly 1`] = `<App />`;

exports[`Provider and App renders <Provider/> correctly 2`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Connect(Header)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/login"
    />
    <Route
      component={[Function]}
      path="/signup"
    />
    <Route
      component={[Function]}
      path="/articles/page/:pageNumber"
    />
    <Route
      component={[Function]}
      path="/article/:articleId"
    />
    <Route
      component={[Function]}
      path="/profile"
    />
    <Route
      component={[Function]}
      path="/profiles/edit"
    />
    <Route
      component={[Function]}
      path="/passwordreset"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/newpassword"
    />
    <Route
      component={[Function]}
      path="/createArticle"
    />
    <Route
      component={[Function]}
      path="/tags"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/profiles/:username"
    />
    <Route
      component={[Function]}
      path="/search"
    />
    <Route
      component={[Function]}
      path="/editarticle/:articleId"
    />
  </Switch>
  <_class
    options={Object {}}
  />
  <Connect(Footer) />
</Router>
`;
